source/agora/consensus/Quorum.d

```diff
@@ -0,0 +1,1118 @@
+        sort(nodes);
+
+        res ~= format("\n%s: QuorumConfig(thresh: %s, nodes: %s)",
+            key_to_simple[key],
+            quorum.threshold,
+            nodes);
+    }
+
+    return res;
+}
+
+/// ditto
+private string pretty (QuorumConfig conf)
+{
+    auto nodes = conf.nodes.map!(node => key_to_simple[node]).array;
+    return format("\nQuorumConfig(%s, %s)",
+        conf.threshold,
+        nodes);
+}
+
+// convenience so we can use [n0, n1] when refering to public keys in the tests
+static foreach (node_idx; 0 .. pregen_seeds.length)
+    mixin("PublicKey n%s;".format(node_idx));
+
+#line __LINE__
+
+// convenience so we can use [n0, n1] when refering to public keys in the tests\
+static this ()
+{
+    static foreach (node_idx; 0 .. pregen_seeds.length)
+        mixin("n%1$s = getTestKey(%1$s);".format(node_idx));
+}
+
+#line __LINE__
+
+/*******************************************************************************
+
+    Build the quorum configs for the given enrollments and range of seeds,
+    and return a map of the number of times each node was included in
+    another node's quorum set.
+
+    Returns:
+        the map of node => quorum set inclusion count
+
+*******************************************************************************/
+
+private int[PublicKey] countNodeInclusions (Enrolls, Range)(
+    Enrolls enrolls, Range seeds)
+{
+    int[PublicKey] counts;
+    foreach (rand_seed; seeds)
+    {
+        auto quorums = buildQuorumConfigs(enrolls.expand, rand_seed);
+        verifyQuorumsSanity(quorums);
+
+        // Intersection checks take roughly ~2 minutes for a configuration
+        // of 16 node quorums, and grows exponentially after that.
+        // By default we disable these checks unless overriden with
+        // -version=EnableAllIntersectionChecks
+        version (EnableAllIntersectionChecks)  // enable all checks
+            const bool check_intersections = true;
+        else
+            const bool check_intersections = quorums.length <= 8;
+
+        if (check_intersections)
+        {
+            version (PerformanceTests)
+            {
+                import core.time;
+                import std.datetime;
+                auto sw = StopWatch(Yes.autoStart);
+                scope (exit) writefln("Intersection check for %s %s took %s",
+                    line, rand_idx, sw.peek.seconds);
+            }
+
+            verifyQuorumsIntersect(quorums);
+        }
+
+        foreach (key, quorum; quorums)
+        {
+            foreach (node; quorum.nodes)
+                counts[node]++;
+        }
+    }
+
+    return counts;
+}
+
+/// Generate random seeds by hashing a range of numbers: [0 .. count)
+private auto getRandSeeds (size_t count)
+{
+    // using 'ulong' to get consistent hashes
+    return iota(0, count).map!(idx => hashFull(cast(ulong)idx));
+}
+
+/// ditto
+private string pretty (PublicKey input)
```

Comment:
Noice. Probably something we want to generalize in the long run. I mean Ocean has it, and I think stellar does too.
Something to consider: For easier memorization, you can use a noun and a verb, like what docker does. E.g. "patriot pangolin" and "mesmerizing door" might stand out more than "Andrew" and "Nathan".

Reply:
ah yes, the gifycat approach.

Or I could use https://github.com/bpfkorea/ocean/blob/dev-2020-04-02/src/ocean/io/digest/FirstName.d :D